### PR TITLE
feat: sales history

### LINKED ISSUE #50: Implement sales history server changes

### PR DESCRIPTION
**Pull request by jmulq**
Implement sales history logic

Fix fill buy now input which was causing an issue when trying to fill a buy now the second time.

Closes #50

### CODE REVIEW

File: src/graphql-types/inputs/fillBuyNow.input.ts [hunk 1]
```diff
@@ -3,10 +3,7 @@ import { field, inputType } from '@loopback/graphql'
 @inputType()
 export class FillBuyNowInput {
   @field((type) => String!)
-  assetAddress: string
-
-  @field((type) => String!)
-  assetId: string
+  listingId: string
```
Comment (creativogee) [inline src/graphql-types/inputs/fillBuyNow.input.ts:6]: Isnt this going to be a breaking change

Comment (jmulq) [reply src/graphql-types/inputs/fillBuyNow.input.ts:6]: I updated the front end following this - let me know if I've missed anything in the PR https://github.com/Vault-Hill/vhc-marketplace-app/pull/172/files#diff-abb7caa147a635cda570de888fa3aad8c6c781cc015c122477359480705aaa8b

Comment (creativogee) [reply src/graphql-types/inputs/fillBuyNow.input.ts:6]: I can image some op utilizing the old properties. If this checks in, it is worth assessing the all instance of FillBuyNowInput

Comment (jmulq) [reply src/graphql-types/inputs/fillBuyNow.input.ts:6]: The relevant mutation is only referenced/used in the `SubmitOrder` file which has been changed.

**Review by creativogee — APPROVED** [review]
lgtm

**Pull request merged**